src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -2968,6 +2968,18 @@ def set_up_network_profile(self, mc: ManagedCluster) -> ManagedCluster:
         mc = super().set_up_network_profile(mc)
         network_profile = mc.network_profile
 
+        # network_plugin is typically defaulted to kubenet. AKS-RP is moving
```

Comment:
not only set by SDK, we also set it to kubenet when the lb sku is standard
https://github.com/Azure/azure-cli/blob/dev/src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py#L5887-L5890

Reply:
we should remove that then. but this PR would still work right since setup_network_profile in the cli is called before setup_network_profile from the extension, correct? so if we always clear the value when a user does not specify anything then it takes care of that defaulting you linked.

Reply:
Confused why that code exists at all. Should't the api be doing defaulting there not cli? Might be worth stating a conversation with Pengfei's team about. 

    # Attention: RP would return UnexpectedLoadBalancerSkuForCurrentOutboundConfiguration internal server error
            # if load_balancer_sku is set to basic and load_balancer_profile is assigned.

Reply:
yeah, the function in core cli is called earlier. Also agree with Paul, defaulting should be done by the sever. This piece of code has been around for a long time (4 years+), and perhaps the server did not have a suitable default logic at the time.